dispatchers.go

```diff
@@ -164,14 +166,32 @@ func add(b *Builder, args []string, attributes map[string]bool, flagArgs []strin
 			return fmt.Errorf("ADD only supports the --chmod=<permissions> and the --chown=<uid:gid> flag")
 		}
 	}
-	b.PendingCopies = append(b.PendingCopies, Copy{Src: args[0:last], Dest: dest, Download: true, Chown: chown, Chmod: chmod})
+	var files []File
+	for _, heredoc := range heredocs {
+		command := parseCommandForHeredoc(original)
+		content := heredoc.Content
+		if heredoc.Chomp {
+			content = buildkitparser.ChompHeredocContent(content)
+		}
+		file := File{
+			Data:    content,
+			Name:    heredoc.Name,
+			Command: command,
+		}
+		files = append(files, file)
+	}
+	isheredoc := false
+	if len(files) > 0 {
+		isheredoc = true
+	}
```

Comment:
This chunk is copied verbatim in three places.  It should be a helper function that also takes the `command` as a parameter, since each caller already knows exactly what its value should be.

Reply:
Oh, `command` is not the instruction.

Reply:
Fixed.